### PR TITLE
Create a new singleton metric with an API_NAME tag

### PR DESCRIPTION
**Pull request by andrew4699**
# Description

Currently we emit metrics with the API name in the metric name, for example `polaris.principals.getPrincipal`. This is difficult to query in i.e. Grafana; metric names are not intended to be "dynamic".

This PR adds a new equivalent metric `polaris.TimedApi` with the `API_NAME` tag.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] Created a test for TimedApplicationEventListener

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue

### CODE REVIEW

File: polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java [hunk 1]
```diff
@@ -38,13 +40,21 @@
 @Provider
 public class TimedApplicationEventListener implements ApplicationEventListener {
 
+  public static final String METRIC_NAME = "polaris.TimedApi";
```
Comment (collado-mike) [inline polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java:43]: Does this get prefixed to the existing metric name? E.g., currently we publish `polaris.CatalogApi.commitTransaction`. Will it now be `polaris.TimedApi.polaris.CatalogApi.commitTransaction`?

File: polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java [hunk 2]
```diff
@@ -38,13 +40,21 @@
...
+  public static final String TAG_API_NAME = "API_NAME";
```
Comment (collado-mike) [inline polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java:44]: I think tag names should be lowercase

Comment (andrew4699) [reply polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java:44]: I agree, but we currently uppercase the realm/status tags: https://github.com/apache/polaris/blob/main/polaris-core/src/main/java/org/apache/polaris/core/monitor/PolarisMetricRegistry.java#L45-L46

I'm not sure if we'd want to leave the old tags uppercase or if we want to change them, how we should handle backward compatibility and eventually full deprecation.

File: polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java (hunk 1 above)
Comment (andrew4699) [reply polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java]: We still emit the former metric and a new one that's just called `polaris.TimedApi` for all API endpoints.

Comment (collado-mike) [reply polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java]: Can we just make this `polaris.api`? I propose we go with the naming convention of all lowercase and underscore separated - I feel that's a much more common metric naming convention than the CamelCase we use here. I think we can deprecate the old metric names and eventually get rid of them.

File: polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java (hunk 2 above)
Comment (collado-mike) [reply polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java:49]: Lets use lowercase for the new metrics and deprecate the old ones

File: polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java (hunk 1 above)
Comment (andrew4699) [reply polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java]: Updated!

File: polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java (hunk 2 above)
Comment (andrew4699) [reply polaris-service/src/main/java/org/apache/polaris/service/TimedApplicationEventListener.java]: Done! The new tags only emit the `snake_case` version and the old ones emit both. Added lots of comments about deprecation and beefed up the tests.

**Review by eric-maynard — APPROVED** [review]

**Review by eric-maynard — APPROVED** [review]

**Review by eric-maynard — APPROVED** [review]

**Pull request merged**